sdk/iot/core/src/az_iot_core.c

```diff
@@ -10,3 +10,30 @@
 
 #include <_az_cfg.h>
 
+AZ_NODISCARD az_span az_span_token(az_span source, az_span delimiter, az_span* out_remainder)
+{
+  AZ_PRECONDITION_VALID_SPAN(delimiter, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_remainder);
+
+  if (az_span_length(source) == 0)
+  {
+    return AZ_SPAN_NULL;
+  }
+  else
+  {
+    int32_t index = az_span_find(source, delimiter);
+
+    if (index != -1)
+    {
+      *out_remainder = az_span_slice(source, index + az_span_length(delimiter), az_span_length(source));
```

Comment:
in the docs it says:
> If the position after `delimiter` is the end of `source`, `out_remainder` is set to a NULL/empty #az_span.

Is that true here? Shouldn't there be a check for `index + length(delim) == length(source)`